test/e2e/installplan_e2e_test.go

```diff
@@ -2998,6 +2999,155 @@ var _ = Describe("Install Plan", func() {
+							},
+							Conditions: []operatorsv1alpha1.BundleLookupCondition{
+								{
+									Type:               operatorsv1alpha1.BundleLookupPending,
+									Status:             corev1.ConditionTrue,
+									Reason:             "JobIncomplete",
+									Message:            "unpack job not completed",
+									LastTransitionTime: &now,
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		AfterEach(func() {
+			Expect(ctx.Ctx().Client().Delete(context.Background(), ns)).To(Succeed())
+		})
+
+		It("should show an error on the bundlelookup condition for a non-existent bundle image", func() {
+			// Create an InstallPlan status.bundleLookups.Path specified for a non-existent bundle image
+			ip.Status.BundleLookups[0].Path = "quay.io/foo/bar:v0.0.1"
+
+			outIP := ip.DeepCopy()
+
+			Expect(ctx.Ctx().Client().Create(context.Background(), outIP)).To(Succeed())
+
+			// The status gets ignored on create so we need to update it else the InstallPlan sync ignores
+			// InstallPlans without any steps or bundle lookups
+			outIP.Status = ip.Status
+			Expect(ctx.Ctx().Client().Status().Update(context.Background(), outIP)).To(Succeed())
+
+			// The InstallPlan's status.bundleLookup.conditions should have a BundleLookupPending condition
+			// with the container status from unpack pod that mentions an image pull failure for the non-existent
+			// image, e.g ErrImagePull or ImagePullBackOff
+			Eventually(
+				func() (string, error) {
+					err := ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(outIP), outIP)
+					if err != nil {
+						return "", err
+					}
+					for _, bl := range outIP.Status.BundleLookups {
+						for _, cond := range bl.Conditions {
+							if cond.Type != operatorsv1alpha1.BundleLookupPending {
+								continue
+							}
+							return cond.Message, nil
+						}
+					}
+					return "", fmt.Errorf("%s condition not found", operatorsv1alpha1.BundleLookupPending)
+				},
+				1*time.Minute,
+			).Should(ContainSubstring("ErrImagePull"))
+		})
+
+		It("should timeout and fail the InstallPlan for an invalid bundle image", func() {
+			// Create an InstallPlan status.bundleLookups.Path specified for an invalid bundle image
+			ip.Status.BundleLookups[0].Path = "alpine:3.13"
+
+			outIP := ip.DeepCopy()
+
+			Expect(ctx.Ctx().Client().Create(context.Background(), outIP)).To(Succeed())
+
+			// The status gets ignored on create so we need to update it else the InstallPlan sync ignores
+			// InstallPlans without any steps or bundle lookups
+			outIP.Status = ip.Status
+			Expect(ctx.Ctx().Client().Status().Update(context.Background(), outIP)).To(Succeed())
+
+			// The InstallPlan should be Failed due to the bundle unpack job timeout
+			// We wait for some time over the default --bundle-unpack-timeout (i.e ActiveDeadlineSeconds) so that the Job can fail
+			defaultBundleUnpackTimeout := 10 * time.Minute
+			waitFor := defaultBundleUnpackTimeout + 30*time.Second
```

Comment:
Can we add an optional timeout field to the InstallPlan spec (using the command-line flag as a default) to make this easier to test?

Reply:
As discussed offline, while a new `spec.bundleUnpackTimeout` won't be back-portable we can specify a per InstallPlan bundle unpack timeout via an annotation for now, e.g:
```YAML
apiVersion: operators.coreos.com/v1alpha1
kind: InstallPlan
metadata:
  annotations:
    olm/bundle-unpack-timeout: 2m
```

That way we can set a shorter timeout for this test.

And we can potentially graduate the annotation to a spec field later on.

Reply:
Also I just realized I have this timeout on the wrong test. It should be on the other one for `should show an error on the bundlelookup condition for a non-existent bundle image`.
That's where we need to wait to expire the 10m `ActiveDeadlineSeconds` because the pods stay pending.

Here we can simply wait for the `BackoffLimit` of 3 retries to fail i.e (10s+20s+40s) or > 1m10s.

I'll update that as well.